src/commands/scanner/rule/list.ts

```diff
@@ -59,58 +61,71 @@ export default class List extends SfdxCommand {
     })
   };
 
-  /**
-   * Get high-level information about the rules that match the provided filter criteria.
-   * @param {string[]|null} cats - If non-null and non-empty, only rules with a matching category tag will be returned.
-   * @param {string[]|null} rulesets - If non-null and non-empty, only rules in the given ruleset will be returned.
-   * @param {string|null} sev - If non-null, only rules of the specified severity will be returned.
-   * @param {string[]|null} langs - If non-null and non-empty, only rules targeting the specified languages will be returned.
-   * @param {AuthorFilter} author - Only rules authored by the specified author will be returned.
-   * @returns {Promise<AnyJson[]|string>} Resolves to a list of rules, or rejects with an error message.
-   * @private
-   */
-  private async getRules(cats : string[], rulesets : string[], sev : string, langs : string[], author : AuthorFilter) : Promise<AnyJson[]|string> {
-    let rules = [
-      {
-        name: 'Rule 1',
-        categories: ['Best Practice', 'Code Styling'],
-        rulesets: ['core/best-practice'],
-        languages: ['JS', 'Apex', 'Java'],
-        author: 'Salesforce'
-      },
-      {
-        name: 'Rule 2',
-        categories: ['Security', 'XSS'],
-        rulesets: ['lib/security'],
-        languages: ['JS'],
-        author: 'Doofenshmirtz Evil Inc'
-      }
-    ];
-    return new Promise((res, rej) => {
-      setTimeout(() => {
-        res(rules);
-      }, 2500);
-    });
+  public async run() : Promise<AnyJson> {
+    // We'll have a bunch of promises to load rules from their source files.
+    let pmdPromise = this.getPmdRules();
+
+    // Once all of those promises are done, we'll filter all of the results.
+    return Promise.all([pmdPromise])
+      .then((res : any[]) => {
+        let filteredPmdRules = res[0].filter((rule) => {return this.ruleSatisfiesFilterConstraints(rule);});
+        this.ux.table(filteredPmdRules, columns);
+        return filteredPmdRules;
+      }, (rej : string) => {
+        throw new SfdxError(rej);
+      })
   }
 
-  public async run(): Promise<AnyJson> {
-    const cats = this.flags.category;
-    const rulesets = this.flags.ruleset;
-    const sev = this.flags.severity;
-    const langs = this.flags.language;
-    const author = this.flags.standard ? AuthorFilter.StandardOnly : this.flags.custom ? AuthorFilter.CustomOnly : AuthorFilter.All;
+  private async getPmdRules() : Promise<AnyJson> {
+    // PmdCatalogWrapper is a layer of abstraction between the plugin and PMD, facilitating code reuse and other goodness.
+    return new PmdCatalogWrapper().getCatalog()
```

Comment:
Why not async/await?

Plus, are you handling errors properly here? I don't see a catch() after your then()